features/admin.server-configurations.v1/forms/username-recovery-form.tsx

```diff
@@ -100,34 +107,92 @@ export const UsernameRecoveryConfigurationForm: FunctionComponent<UsernameRecove
      * @param values - Form values.
      * @returns Sanitized form values.
      */
-    const getUpdatedConfigurations = (value: boolean) => {
-        const data: { [ key: string ]: unknown } = {
-            [ GovernanceConnectorUtils.decodeConnectorPropertyName(
-                ServerConfigurationsConstants.USERNAME_RECOVERY_ENABLE) ]: value.toString()
+    const getUpdatedConfigurations = (values: Record<string, any>) => {
+
+        const data: UsernamePasswordRecoveryFormUpdatableConfigsInterface = {
+            "Recovery.Notification.Username.Email.Enable":
+                values.enableEmailBasedRecovery ?? initialConnectorValues?.enableEmailBasedRecovery,
+
+            "Recovery.Notification.Username.SMS.Enable":
+                values.enableSMSBasedRecovery ?? initialConnectorValues?.enableSMSBasedRecovery
         };
 
         return data;
     };
 
+    if (!initialConnectorValues) {
+        return null;
+    }
+
     return (
         <div className={ "connector-form" }>
             <Form
                 id={ FORM_ID }
-                onSubmit={ null }
+                onSubmit={ (values: Record<string, any>) => onSubmit(getUpdatedConfigurations(values)) }
```

Comment:
Can we use `boolean` instead of `any` here?

Reply:
Fixed by https://github.com/wso2/identity-apps/pull/7004/commits/a81d0a35e1e7b077862d1d44dfde06a60da22794